zerver/lib/actions.py

```diff
@@ -3926,24 +3927,29 @@ def __init__(self, msg: Text, errors: List[Tuple[Text, str]], sent_invitations:
         self.errors = errors  # type: List[Tuple[Text, str]]
         self.sent_invitations = sent_invitations  # type: bool
 
-def do_invite_users(user_profile: UserProfile,
-                    invitee_emails: SizedTextIterable,
-                    streams: Iterable[Stream],
-                    invite_as_admin: Optional[bool]=False,
-                    body: Optional[str]=None) -> None:
+def check_invite_limit(user: UserProfile, num_invitees: int) -> None:
     if settings.OPEN_REALM_CREATION:
         # Discourage using invitation emails as a vector for carrying spam
-        sent_invites = Confirmation.objects.filter(
-            realm=user_profile.realm,
-            date_sent__gte=timezone_now() - datetime.timedelta(days=1),
-            type=Confirmation.INVITATION).count()
-        if len(invitee_emails) + sent_invites > user_profile.realm.max_invites:
+        sent_invites = RealmCount.objects.filter(
+            realm=user.realm,
+            property='invites_sent::day',
+            end_time__gte=timezone_now() - datetime.timedelta(days=1)
+        ).aggregate(Sum('value'))['value__sum']
+        if sent_invites is None:
+            sent_invites = 0
+        if num_invitees + sent_invites > user.realm.max_invites:
             raise InvitationError(
-                _("You do not have enough remaining invites; "
-                  "try again with fewer emails, or contact %s. "
+                _("You do not have enough remaining invites. "
+                  "Please contact %s to have your limit raised. "
                   "No invitations were sent." % (settings.ZULIP_ADMINISTRATOR)),
                 [], sent_invitations=False)
 
+def do_invite_users(user_profile: UserProfile,
+                    invitee_emails: SizedTextIterable,
+                    streams: Iterable[Stream],
+                    invite_as_admin: Optional[bool]=False,
+                    body: Optional[str]=None) -> None:
+    check_invite_limit(user_profile, len(invitee_emails))
     validated_emails = []  # type: List[Text]
     errors = []  # type: List[Tuple[Text, str]]
     skipped = []  # type: List[Tuple[Text, str]]
```

Comment:
Small style bit: I'd put blank lines before and after this `check_invite_limit`. Otherwise (a) it looks grouped with the `validated_emails = []` etc. initializations, and (b) it gets kind of camouflaged visually with the parameter list.